client/src/Player.lua

```diff
@@ -97,36 +97,7 @@ function Player:createStackFromSettings(match, which)
   self.stack = Stack(args)
   -- so the stack can draw player information
   self.stack.player = self
-  self.stack.battleAnimations = {}
-  for name, set in pairs(characters[self.stack.character].battleAnimations) do
-    local anim = set:clone()
-    anim:stop()
-    anim:setSwitchFunction(
-      function (s, state)
-        local switch
-        local finish = false
-        if self.stack.match.ended then
-          switch = (self.stack.game_over_clock <= 0 and "win") or "lose"
-        elseif state == "hurt" or self.stack.shake_time > 0 then
-          switch = "hurt"
-        elseif state == "attack" then
-          switch = "attack"
-        elseif (self.stack.danger)then
-          switch = "danger"
-          finish = anim.currentAnim ~= "normal"
-        else
-          switch = "normal"
-          finish = true
-        end
-        if anim.animations[switch] ~= nil then
-          s:switchAnimation(switch, finish)
-        end
-      end
-    )
-    self.stack.battleAnimations[name] = anim
-    self.stack:connectSignal("hurt", self.stack.battleAnimations[name], self.stack.battleAnimations[name].switchFunction)
-    self.stack:connectSignal("attackSent", self.stack.battleAnimations[name], self.stack.battleAnimations[name].switchFunction)
-  end
+
```

Comment:
Setting up the animations per stack to enable mirrors is not something that should happen in Player:createStackFromSettings. That function is very specifically of mapping player settings to arguments to pass into the Stack constructor. I think you can agree that setting up battle animations has nothing to do with the Player so the setup code shouldn't be in that class but rather inside Stack or Character so I moved it into a function on Character that takes the stack as an argument and performs the cloning.